lib/system/extra/cmd_streamer.ex

```diff
@@ -0,0 +1,143 @@
+defmodule System.Extra.CmdStreamer do
+  @moduledoc """
+  Module allowing Enumerables to be constructed around Shell commands.
+
+  """
+
+  alias __MODULE__
+
+
+
+  #########################################################
+  # Types
+  #########################################################
+
+  @opaque t :: %__MODULE__{
+    port: port,
+    queue: :queue.queue,
+    receiver: pid,
+    running: boolean,
+    pending_demand: non_neg_integer,
+  }
+
+  defstruct [
+    port: nil,
+    queue: :queue.new(),
+    receiver: nil,
+    running: true,
+    pending_demand: 0,
+  ]
+
+
+
+  #########################################################
+  # Public API
+  #########################################################
+
+  @doc """
+  Starts the `CmdStreamer` enumerable by wrapping `cmd` and forwarding `args` to it.
+
+  """
+  @spec start(binary, [binary]) :: pid
+  def start(cmd, args) do
+    receiver =
+      self()
+
+    cmd_string =
+      [cmd | args] |> Enum.join(" ")
+
+    spawn(fn ->
+      port =
+        Port.open({:spawn, cmd_string}, [{:line, 20_000}, :binary, :exit_status])
```

Comment:
Hmm this doesn't actually error on lines over 20,000. From the docs:

> Lines longer than this are delivered in more than one message, with Flag set to noeol for all
> but the last message. If end of file is encountered anywhere else than immediately following
> a newline sequence, the last line is also delivered with Flag set to noeol. Otherwise lines are 
> delivered with Flag set to eol.